components/calendar/src/islamic.rs

```diff
@@ -741,81 +881,35 @@ impl<A: AsCalendar<Calendar = IslamicUmmAlQura>> DateTime<A> {
     }
 }
 
-impl IslamicUmmAlQura {
-    fn fixed_from_saudi_islamic(i_date: IslamicUmmAlQuraDateInner) -> RataDie {
-        calendrical_calculations::islamic::fixed_from_saudi_islamic(
-            i_date.0.year,
-            i_date.0.month,
-            i_date.0.day,
-        )
-    }
-
-    fn saudi_islamic_from_fixed(date: RataDie) -> Date<IslamicUmmAlQura> {
-        let (y, m, d) = calendrical_calculations::islamic::saudi_islamic_from_fixed(date);
-
-        debug_assert!(Date::try_new_ummalqura_date(
-            y,
-            m,
-            d,
-            IslamicUmmAlQura::new_always_calculating()
-        )
-        .is_ok());
-        Date::from_raw(
-            IslamicUmmAlQuraDateInner(ArithmeticDate::new_unchecked(y, m, d)),
-            IslamicUmmAlQura,
-        )
-    }
-
-    fn year_as_islamic(year: i32) -> types::FormattableYear {
-        types::FormattableYear {
-            era: types::Era(tinystr!(16, "islamic")),
-            number: year,
-            cyclic: None,
-            related_iso: None,
-        }
-    }
-}
-
 /// The inner date type used for representing [`Date`]s of [`IslamicCivil`]. See [`Date`] and [`IslamicCivil`] for more details.
 
 #[derive(Copy, Clone, Debug, Hash, Eq, PartialEq, PartialOrd, Ord)]
 pub struct IslamicCivilDateInner(ArithmeticDate<IslamicCivil>);
 
 impl CalendarArithmetic for IslamicCivil {
-    type YearInfo = ();
-
-    fn month_days(year: i32, month: u8, _data: ()) -> u8 {
-        match month {
-            1 | 3 | 5 | 7 | 9 | 11 => 30,
-            2 | 4 | 6 | 8 | 10 => 29,
-            12 if Self::is_leap_year(year, ()) => 30,
-            12 => 29,
-            _ => 0,
-        }
+    type YearInfo = IslamicYearInfo;
+
+    fn month_days(_year: i32, month: u8, year_info: IslamicYearInfo) -> u8 {
+        year_info.packed_data.days_in_month(month)
     }
```

Comment:
This calendar seems to have a nice formula for month lengths. Why make that part data-driven? I thought we would only make expensive computations be data-driven.

Reply:
Ah, now I remember. We only wanted to optimize UAQ and observational.